bee-protocol/src/tangle/mod.rs

```diff
@@ -191,6 +197,48 @@ impl MsTangle {
                 .unwrap_or(false)
         }
     }
+
+    pub fn otrsi(&self, hash: &Hash) -> Option<MilestoneIndex> {
+        if let Some(sep) = self.solid_entry_points.get(hash) {
+            Some(*sep.value())
+        } else {
+            match self.get_metadata(hash) {
+                Some(metadata) => metadata.otrsi(),
+                None => None,
+            }
+        }
+    }
+
+    pub fn ytrsi(&self, hash: &Hash) -> Option<MilestoneIndex> {
+        if let Some(sep) = self.solid_entry_points.get(hash) {
+            Some(*sep.value())
+        } else {
+            match self.get_metadata(hash) {
+                Some(metadata) => metadata.ytrsi(),
+                None => None,
+            }
+        }
+    }
+
+    pub fn insert_tip(&self, tail: Hash, trunk: Hash, branch: Hash) {
+        let mut pool = self.tip_pool.lock().unwrap();
+        pool.insert(tail, trunk, branch);
```

Comment:
Can we reduce to one instruction ? Same for the following methods. I find it more elegant :)

Reply:
Can we actually? "When the guard goes out of scope, the mutex will be unlocked." If i remeber correctly I got a warning about that

Reply:
I don't know, I'm asking :)

Reply:
Okey let me check again in detail 👍

Reply:
Things don't get dropped until the end of an expression, so yes, we can change the expressions.